pkg/controllers/dynakube/injection/reconciler.go

```diff
@@ -86,13 +86,31 @@ func (r *Reconciler) Reconcile(ctx context.Context) error {
 		defer r.cleanup(ctx)
 	} else {
 		dkMapper := r.createDynakubeMapper(ctx)
-		if err := dkMapper.MapFromDynakube(); err != nil {
+
+		oaNames, meNames, err := dkMapper.MapFromDynakube()
```

Comment:
why is this not done in the mapper,

here all you would need to say is `err := dkMapper.MapFromDynakube()`

all the rest with the conditions should be in the mapper, as it should "own" those conditions